src/network/core/tcp.h

```diff
@@ -82,10 +82,12 @@ class TCPConnecter {
 		RESOLVING,  ///< The hostname is being resolved (threaded).
 		FAILURE,    ///< Resolving failed.
 		CONNECTING, ///< We are currently connecting.
+		HAS_RESULT, ///< Our connecter is finished and "result_socket" contains the result (TCPServerConnecter only).
```

Comment:
`HAS_RESULT` is a quite non-descript status name. The name itself doesn't give much of a clue.

Maybe GC_FINISHED? As in: "The Game Coordinator has finished its connection attempts, and set the socket to use which could be INVALID_SOCKET in case of a failure."
Potentially prefixing the others (except INIT) with DC_... to make it clearer those are used for Direct (Connect).

Alternatively add a CONNECTED state instead, with a SetFailure() and SetConnected(SOCKET socket) so the code with SetResult(INVALID_SOCKET) can become SetFailure().
In CheckActivity for TCPServerConnector you can then read the status; with CONNECTED -> OnSuccess, FAILURE -> OnFailure, rest: return false. I would even set the state to CONNECTING after calling ConnectToServer, so when debugging the status shows CONNECTING instead of INIT.